app.rb

```diff
@@ -58,4 +62,71 @@ def add_music_album
     @albums << new_music_album
     @genres << new_music_album.genre
   end
+
+  def get_input(request_text)
+    print "#{request_text}: "
+    gets.chomp
+  end
+
+  def add_book(newbook)
+    newbook_instance = Book.new(*newbook)
+    hash = {
+      'id' => newbook_instance.id,
+      'name' => newbook_instance.name,
+      # 'genre' => newbook_instance.genre,
+      'label' => newbook_instance.label,
+      'publish_date' => newbook_instance.publish_date,
+      'publisher' => newbook_instance.publisher,
+      'cover_state' => newbook_instance.cover_state
+    }
+    @books << hash
+  end
+
+  def add_book_details
+    puts 'Enter book details'
+    id = Random.rand(1..500)
+    name = get_input('Input book name')
+    # genre = get_input('Input book genre')
```

Comment:
- [ ] (Optional). Please remove the unnecessary line of code, instead of commenting out.